panther_lights/test/test_led_animations_queue.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <yaml-cpp/yaml.h>
+
+#include <rclcpp/time.hpp>
+
+#include <panther_lights/led_animations_queue.hpp>
+#include <panther_lights/led_segment.hpp>
+
+class TestLEDAnimationsQueue : public testing::Test
+{
+public:
+  TestLEDAnimationsQueue();
+  ~TestLEDAnimationsQueue() {}
+
+  panther_lights::LEDAnimation CreateLEDAnimation(
+    const std::string & name, const std::uint8_t priority,
+    const rclcpp::Time & init_time = rclcpp::Time(0));
+
+protected:
+  std::shared_ptr<panther_lights::LEDAnimationsQueue> led_anim_queue_;
+  std::unordered_map<std::string, std::shared_ptr<panther_lights::LEDSegment>> segments_;
+
+  const std::size_t max_queue_size_ = 5;
+};
+
+TestLEDAnimationsQueue::TestLEDAnimationsQueue()
+{
+  auto segment_1_desc = YAML::Load("{channel: 1, led_range: 0-10}");
+  auto segment_2_desc = YAML::Load("{channel: 2, led_range: 0-10}");
+  segments_.emplace(
+    "segment_1", std::make_shared<panther_lights::LEDSegment>(segment_1_desc, 50.0));
+  segments_.emplace(
+    "segment_2", std::make_shared<panther_lights::LEDSegment>(segment_2_desc, 50.0));
+
+  led_anim_queue_ = std::make_shared<panther_lights::LEDAnimationsQueue>(5);
+}
+
+panther_lights::LEDAnimation TestLEDAnimationsQueue::CreateLEDAnimation(
+  const std::string & name, const std::uint8_t priority, const rclcpp::Time & init_time)
+{
+  panther_lights::AnimationDescription anim_desc;
+  anim_desc.segments = {"segment_1", "segment_2"};
+  anim_desc.type = "panther_lights::ImageAnimation";
+  anim_desc.animation =
+    YAML::Load("{image: $(find panther_lights)/animations/triangle01_red.png, duration: 2.0}");
+
+  panther_lights::LEDAnimationDescription led_anim_desc;
+  led_anim_desc.id = 0;
+  led_anim_desc.name = name;
+  led_anim_desc.priority = priority;
+  led_anim_desc.timeout = 10.0;
+  led_anim_desc.animations = {anim_desc};
+
+  return panther_lights::LEDAnimation(led_anim_desc, segments_, init_time);
+}
+
+TEST_F(TestLEDAnimationsQueue, Put)
+{
+  EXPECT_TRUE(led_anim_queue_->Empty());
```

Comment:
1. If the queue is not empty, there is no sense in checking later macros. I would use `ASSERT_TRUE`.
2. Not sure if it makes sense to check it at all since we create a fresh instance of `TestLEDAnimationsQueue` for each test case. Am I missing sth 🤔 

This comment applies to all similar scenarios in this test file.

Reply:
I have no idea why i put it there